frontend/src/metabase/entities/databases/forms.js

```diff
@@ -260,6 +260,14 @@ function getEngineFormFields(engine, details, id) {
         helperText: field["helper-text"],
         visibleIf: field["visible-if"],
         treatBeforePosting: field["treat-before-posting"],
+        getError: (errors = {}) => {
+          const fieldError = errors[field.name];
+          if (fieldError) {
+            return {
+              details: { [field.name]: fieldError },
+            };
+          }
+        },
```

Comment:
Since BE sends the error response as
```json
{
  "message": "Looks like your Username is incorrect.",
  "errors": {
    "user": "check your username"
  }
}
```

I need to rewrite the errors to `redux-form` as
```js
{
  details: {
    user: "check your username"
  }
}
```

Otherwise, the inline field error won't show up. I've also tried to remove the `details.` from the field name, but there are a lot of other places that depends on `details` so, I don't want to do that yet.

The other approach might be having BE sent the the error like this, so there would be no FE code required. But I'm not sure if this would be a good idea or not.

```json
{
  "message": "Looks like your Username is incorrect.",
  "errors": {
    "detail": {
      "user": "check your username"
    }
  }
}
```

Any thoughts @ranquild @metamben ?

Reply:
@WiNloSt Setup uses the following code to handle error messages for `details` fields  https://github.com/metabase/metabase/blob/62885b4084b7b449c77ed5fd39fd3bce5c365afc/frontend/src/metabase/setup/components/DatabaseStep/DatabaseStep.tsx#L202

Reply:
> @WiNloSt Setup uses the following code to handle error messages for `details` fields
> 
> https://github.com/metabase/metabase/blob/62885b4084b7b449c77ed5fd39fd3bce5c365afc/frontend/src/metabase/setup/components/DatabaseStep/DatabaseStep.tsx#L202

Awesome! Thanks for pointing that out. I think I can finally drop all the form API extensions.